data_model/src/expression.rs

```diff
@@ -457,6 +490,51 @@ impl<T: Into<Value>> From<T> for ExpressionBox {
     }
 }
 
+/// Deserialize [`Expression`] with possibility to omit [`Raw`](Expression::Raw) tag
+/// as it is the most popular variant.
+///
+/// First will try to deserialize as `Raw` and if it fails will try to deserialize as `Expression`.
```

Comment:
This is normally an anti-pattern, but in this case, fail-fast parsing is the preferred solution.